tests/test_canonicalize.py

```diff
@@ -0,0 +1,216 @@
+import unittest
```

Comment:
Is this a copy of w3lib tests? If yes, please put a link to the original, so that we know where to look for to make an update.